subjects/bowling_score/README.md

```diff
@@ -0,0 +1,106 @@
+## bowling_score
+
+### Instructions
+
+Create a library that implements a `struct` to calculate the score of a player in the game of [Bowling](https://en.wikipedia.org/wiki/Bowling).
+
+The methods of `BowlingGame` will be:
+
+- `roll`: It will take the number of knocked pins and return `Ok(())` if the roll is valid or an `Err(Error::...)` if the roll is invalid.
+- If the pins knocked out are more than the available ones the error will be `NotEnoughPinsLeft`.
+- If the roll happens after the game is complete the error will be `GameComplete`.
+- `score`: It will calculate the score of the player and return `Some(score)` in case of success or `None` if there is still some rolls missing to complete the game.
+
+#### The basic rules of bowling
+
+- The game is divided in 10 frames
+- At the start of each frame 10 pins are placed at the end of the lane
+- Each frame has at maximum 2 rolls
+- `Strike`: You knock 10 pins on the first roll, the game moves directly to the next frame since there is no more pins to knock in this one. You will also get extra points based on the next two rolls.
+- `Spare`: On the second frame you knock all the pins left. You move to the next frame as you would normally but you will get extra points based on the next roll.
+
+Last frame scenario:
+
+- If on the 10th frame you make a `Strike` you will get two extra rolls (2 fill balls).
+- If on the 10th frame you make a `Spare` you will get one extra roll (1 fill ball).
+
+#### The scoring system
+
+- Each pin knocked is one point.
+- If you do a strike (10 pins knocked on first roll of a frame) it will count as 10 + pins knocked in the next two rolls.
```

Comment:
"If you do a strike (knocking down all 10 pins with the first roll of a frame) it will count as 10 points + the number of pins knocked down in the next two rolls."

Reply:
done